comit/src/herc20.rs

```diff
@@ -191,16 +197,18 @@ where
     C: LatestBlock<Block = Block>
         + BlockByHash<Block = Block, BlockHash = Hash>
         + ReceiptByHash
-        + ConnectedNetwork<Network = ChainId>,
+        + TransactionByHash
+        + ConnectedNetwork<Network = ChainId>
+        + GetLogs,
 {
     use crate::btsieve::ethereum::Event;
 
     let event = Event {
         address: params.asset.token_contract,
         topics: vec![
-            Some(Topic(*TRANSFER_LOG_MSG)),
+            Some(*TRANSFER_LOG_MSG),
             None,
-            Some(Topic(deployed.location.into())),
+            Some(deployed.location.into()),
```

Comment:
Didn't you say it would not work for deployed?

Reply:
This is within `watch_for_funded` :)

In this case, `deployed.location` is a parameter of the event we are looking for. More concretely, we are looking for a `TRANSFER` event emitted by the token contract in which the 3rd argument is our HTLC.

In other words: We want a `TRANSFER` event in which our HTLC receives tokens :)